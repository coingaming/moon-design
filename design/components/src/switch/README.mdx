---
name: Switch
menu: Components
route: /components/switch
---

import { Switch } from '@heathmont/sportsbet-components';

# Switch

A switch is simply an alternate appearance for a checkbox that simulates on/off
instead of checked/unchecked.

```jsx react-live
<Switch />
```
