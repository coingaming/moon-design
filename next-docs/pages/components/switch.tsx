import { useState } from 'react';

import { Switch } from '@heathmont/moon-components';

import Preview from '../../components/codePreview/Preview';

const PreviewSwitch = () => {
  const [state, setState] = useState(true);

  return (
    <Preview
      title="Default"
      preview={
        <>
          <Switch
            checked={state}
            onChange={(e) => setState(e.target.checked)}
          />
          ;
        </>
      }
      code={`import { useState } from 'react'
import { Switch } from '@heathmont/moon-components';

const [state, setState] = useState(true);

<Switch checked={state} onChange={e => setState(e.target.checked)} />
`}
    />
  );
};

const PreviewSwitchTheme = () => {
  const [state, setState] = useState(true);

  return (
    <Preview
      title="Dark &amp; Light mode"
      preview={
        <>
          <Switch
            checked={state}
            onChange={(e) => setState(e.target.checked)}
            colorScheme
          />
        </>
      }
      code={`import { useState } from 'react'
import { Switch } from '@heathmont/moon-components';

const [state, setState] = useState(true);

<Switch
  checked={state}
  onChange={(e) => setState(e.target.checked)}
  colorScheme
/>
`}
    />
  );
};

const PreviewSwitchCaption = () => {
  const [state, setState] = useState(true);

  return (
    <Preview
      title="With captions"
      preview={
        <>
          <Switch
            checked={state}
            onChange={(e) => setState(e.target.checked)}
            captionUnchecked="OFF"
            captionChecked="ON"
          />
        </>
      }
      code={`import { useState } from 'react'
import { Switch } from '@heathmont/moon-components';

const [state, setState] = useState(true);

<Switch
  checked={state}
  onChange={e => setState(e.target.checked)}
  captionUnchecked="OFF"
  captionChecked="ON"
/>
`}
    />
  );
};

const PreviewSwitchThemeCaption = () => {
  const [state, setState] = useState(true);

  return (
    <Preview
      title="Dark &amp; Light mode with captions"
      preview={
        <>
          <Switch
            checked={state}
            onChange={(e) => setState(e.target.checked)}
            colorScheme
            captionUnchecked="AM"
            captionChecked="PM"
          />
        </>
      }
      code={`import { useState } from 'react'
import { Switch } from '@heathmont/moon-components';

const [state, setState] = useState(true);

<Switch
      checked={state}
      onChange={e => setState(e.target.checked)}
      colorScheme
      captionUnchecked="AM"
      captionChecked="PM"
    />
`}
    />
  );
};

export default function PageSwitch() {
  return (
    <>
      <section className="mt-8">
        <h1 className="text-5xl font-semibold">Checkbox</h1>
        <p className="text-lg mt-4">Checkbox.</p>
      </section>

      {/* Default */}
      <section className="mt-8">
        <PreviewSwitch />
      </section>

      <section className="mt-8">
        <PreviewSwitchTheme />
      </section>

      <section className="mt-8">
        <PreviewSwitchCaption />
      </section>

      <section className="mt-8">
        <PreviewSwitchThemeCaption />
      </section>
    </>
  );
}
