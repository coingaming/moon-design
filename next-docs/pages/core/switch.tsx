import React from 'react';
import Preview from '../../components/codePreview/Preview';
import ComponentPageDescription from '../../components/ComponentPageDescription';
import PropsTable from '../../components/PropsTable';

import DefaultCode from '../../public/examples/switchTW/codeExamples/DefaultCode';
import CustomBg from '../../public/examples/switchTW/CustomBg';
import Default from '../../public/examples/switchTW/Default';
import Disabled from '../../public/examples/switchTW/Disabled';
import Form from '../../public/examples/switchTW/Form';
import Sizes from '../../public/examples/switchTW/Sizes';
import WithIcons from '../../public/examples/switchTW/WithIcons';
import useExamples from '../../utils/useExamples';

const Example = () => {
  const examples = useExamples('switchTW');
  const examplesCode = useExamples('switchTW/codeExamples');
  return (
    <>
      <ComponentPageDescription title="Switch" isInProgress>
        <p>
          Switch is a control that is used to quickly switch between two
          possible states. Switches are only used for these binary actions that
          occur immediately after the user “flips” the switch. They are commonly
          used for “on/off” switches.
        </p>
        <p>
          Based on{' '}
          <a
            href="https://headlessui.com/"
            className="transition-colors underline hover:text-piccolo"
            target="_blank"
            rel="noreferrer"
          >
            Headless UI
          </a>
          .
        </p>
      </ComponentPageDescription>
      <Preview
        title="Default"
        preview={<Default />}
        code={examples ? examples.Default : 'Loading'}
      />
      <Preview
        title="Sizes"
        preview={<Sizes />}
        code={examples ? examples.Sizes : 'Loading'}
      />
      <Preview
        title="Disabled"
        preview={<Disabled />}
        code={examples ? examples.Disabled : 'Loading'}
      />
      <Preview
        title="With icons"
        preview={<WithIcons />}
        code={examples ? examples.WithIcons : 'Loading'}
      />
      <Preview
        title="Custom background colour"
        preview={<CustomBg />}
        code={examples ? examples.CustomBg : 'Loading'}
      />
      <Preview
        title="Using with HTML forms"
        preview={<Form />}
        code={examples ? examples.Form : 'Loading'}
      />
      <Preview
        title="Code exemples for full cistomization"
        preview={<DefaultCode />}
        code={examplesCode ? examplesCode.DefaultCode : 'Loading'}
      />
      <PropsTable
        title="Switch props"
        data={[
          {
            name: 'checked',
            type: 'boolean',
            required: false,
            default: 'false',
            description: `Is switch checked/unchecked.`,
          },
          {
            name: 'disabled',
            type: 'boolean',
            required: false,
            default: '_',
            description: 'Set disabled/not disabled switch',
          },
          {
            name: 'name',
            type: 'string',
            required: false,
            default: '-',
            description:
              'The name used when using this component inside a form.',
          },
          {
            name: 'offBgColor',
            type: 'string',
            required: false,
            default: 'bg-beerus',
            description: 'Tailwind class for custom off-state background color',
          },
          {
            name: 'offIcon',
            type: 'JSX.Element | string',
            required: false,
            default: '-',
            description: 'Set icon for unchecked state',
          },
          {
            name: 'onBgColor',
            type: 'string',
            required: false,
            default: 'bg-piccolo',
            description: 'Tailwind class for custom on-state background color',
          },
          {
            name: 'onChange',
            type: '(data) => void',
            required: false,
            default: '-',
            description: `The function to call when the switch is toggled.`,
          },
          {
            name: 'onIcon',
            type: 'JSX.Element | string',
            required: false,
            default: '-',
            description: 'Set icon for checked state',
          },
          {
            name: 'size',
            type: '2xs | xs | sm | string',
            required: false,
            default: 'sm',
            description: `The option's value.`,
          },
          {
            name: 'value',
            type: 'string',
            required: false,
            default: '-',
            description:
              'The value used when using this component inside a form, if it is checked.',
          },
        ]}
      />
    </>
  );
};

export default Example;
