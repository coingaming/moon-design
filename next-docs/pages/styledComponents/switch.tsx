import Preview from '../../components/codePreview/Preview';
import ComponentPageDescription from '../../components/ComponentPageDescription';
import DeprecationWarning from '../../components/facing/DeprecationWarning';
import PropsTable from '../../components/PropsTable';
import Buttons from '../../public/styledExamples/switch/Buttons';
import Default from '../../public/styledExamples/switch/Default';
import Disabled from '../../public/styledExamples/switch/Disabled';
import Rtl from '../../public/styledExamples/switch/Rtl';
import Sizes from '../../public/styledExamples/switch/Sizes';
import Theme from '../../public/styledExamples/switch/Theme';
import useExamples from '../../utils/useExamples';

type Props = {
  isRtl?: boolean;
};

export default function PageSwitch({ isRtl }: Props) {
  const examples = useExamples('switch', 'styled');
  return (
    <>
      <ComponentPageDescription title="Switch" isDeprecated>
        <DeprecationWarning href="/components/switch" name="Switch" />
      </ComponentPageDescription>
      <Preview
        title="Default"
        preview={<Default />}
        code={examples ? examples.Default : 'Loading'}
      />
      <Preview
        title="Sizes"
        preview={<Sizes />}
        code={examples ? examples.Sizes : 'Loading'}
      />
      <Preview
        title="Dark &amp; Light mode"
        preview={<Theme />}
        code={examples ? examples.Theme : 'Loading'}
      />
      <Preview
        title="Disabled"
        preview={<Disabled />}
        code={examples ? examples.Disabled : 'Loading'}
      />
      <Preview
        title="Buttons"
        preview={<Buttons />}
        code={examples ? examples.Buttons : 'Loading'}
      />
      <Preview
        title="Right to left"
        preview={<Rtl />}
        code={examples ? examples.Rtl : 'Loading'}
      />
      <PropsTable
        title="Props"
        data={[
          {
            name: 'id',
            type: 'string',
            required: false,
            default: '-',
            description: `Unique element's identifier`,
          },
          {
            name: 'checked',
            type: 'boolean',
            required: false,
            default: 'false',
            description: `Is switch checked/unchecked`,
          },
          {
            name: 'onChange',
            type: 'React.ChangeEvent',
            required: false,
            default: '-',
            description: `Event that happens when you click on the switch`,
          },
          {
            name: 'captionUnchecked',
            type: 'string',
            required: false,
            default: '-',
            description: `Displayed text when unchecked`,
          },
          {
            name: 'captionChecked',
            type: 'string',
            required: false,
            default: '-',
            description: `Displayed text when checked`,
          },
          {
            name: 'isRtl',
            type: 'boolean',
            required: false,
            default: 'false',
            description: `-`,
          },
        ]}
      />
    </>
  );
}
