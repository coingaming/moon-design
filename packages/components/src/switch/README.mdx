---
name: Switch
menu: Components
route: /components/switch
---

import { Switch } from '@heathmont/sportsbet-components';

# Switch

An alternate checkbox appearance for simulating on/off state.

```jsx react-live
<Switch />
```

An additional variant of the `Switch` can be used to toggle the app's
`colorScheme` between 'light' and 'dark'.

```jsx react-live
<Switch colorScheme />
```

## Captions

For extra context, captions can be appended to the unchecked and checked
positions of the switch.

```jsx react-live
<Switch captionUnchecked="ON" captionChecked="OFF" />
```

```jsx react-live
<Switch colorScheme captionUnchecked="AM" captionChecked="PM" />
```

## State

Any manipulation of state should be handled via `onChange`. Below is an example
with React Hooks:

```jsx react-live
() => {
  const [state, setState] = React.useState(true);

  return <Switch checked={state} onChange={e => setState(e.target.checked)} />;
};
```
