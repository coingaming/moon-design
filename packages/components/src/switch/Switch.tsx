import React from 'react';
import styled from 'styled-components';
import { rem, uniqueId, inlineSvg } from '@heathmont/moon-utils';
import hideVisually from 'polished/lib/mixins/hideVisually';

import IconMoon from '../private/icons/IconMoon';
import IconSun from '../private/icons/IconSun';
import Inline from '../inline/Inline';
import Caption from '../caption/Caption';

const switchWidthProperty = '--switch-width';
const switchWidth = `var(${switchWidthProperty})`;

const switchHeightProperty = '--switch-height';
const switchHeight = `var(${switchHeightProperty})`;

type SliderColorScheme = {
  colorScheme?: boolean;
};

const Slider = styled.span<SliderColorScheme>(
  ({
    colorScheme,
    theme: { color, colorScheme: themeColorScheme, space, transitionDuration },
  }) => ({
    position: 'absolute',
    width: switchWidth,
    height: switchHeight,
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    cursor: 'pointer',
    backgroundColor: color.gohan[100],
    backgroundImage: colorScheme
      ? `${inlineSvg(<IconMoon />)}, ${inlineSvg(<IconSun />)}`
      : undefined,
    backgroundPosition: `left ${rem(space.small)} center, right ${rem(
      space.small
    )} center`,
    backgroundRepeat: 'no-repeat',
    borderRadius: `calc(${switchWidth} + ${rem(2)})`,
    transitionProperty: 'background-color',
    transitionDuration: `${transitionDuration.slow}s`,
    transitionTimingFunction: 'ease',
    '&::before': {
      '--switch-indicator-size': `calc(${switchHeight} - ${rem(space.small)})`,
      content: '""',
      position: 'absolute',
      width: 'var(--switch-indicator-size)',
      height: 'var(--switch-indicator-size)',
      left: rem(space.xsmall),
      bottom: rem(space.xsmall),
      backgroundColor: themeColorScheme ? color.goku[100] : color.trunks[100],
      borderRadius: '50%',
      transition: 'inherit',
      transitionProperty: 'background-color, transform',
      willChange: 'transform',
    },
  })
);

const Label = styled.label(({ theme: { space } }) => ({
  [switchWidthProperty]: rem(space.large * 2),
  [switchHeightProperty]: rem(space.large),
  position: 'relative',
  display: 'inline-block',
  width: switchWidth,
  height: switchHeight,
  flexShrink: 0,
}));

const Input = styled.input<SliderColorScheme>(
  ({ colorScheme, theme: { color } }) => ({
    ...hideVisually(),
    [`&:checked + ${Slider}`]: {
      backgroundColor: colorScheme ? undefined : color.piccolo[100],
      '&::before': {
        backgroundColor: colorScheme ? undefined : color.goten[100],
        transform: `translateX(calc(${switchWidth} / 2))`,
      },
    },
  })
);

Input.defaultProps = {
  type: 'checkbox',
};

const SwitcherCaption = styled(Caption as any)<{ checked: boolean }>(
  ({ checked, theme: { color, transitionDuration } }) => [
    { transition: `color ${transitionDuration.default}s` },
    checked && {
      color: checked ? color.trunks[100] : color.bulma[100],
    },
  ]
);

type HTMLInputProps = React.InputHTMLAttributes<HTMLInputElement>;

export type SwitchProps = {
  checked?: boolean;
  className?: string;
  captionUnchecked?: string;
  captionChecked?: string;
  id?: HTMLInputProps['id'];
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
} & SliderColorScheme;

const Switch: React.FC<SwitchProps> = ({
  className,
  captionChecked,
  captionUnchecked,
  checked = false,
  colorScheme,
  id,
  ...props
}) => {
  const autoId = id || `Switch-${uniqueId()}`;

  const labelProps = {
    className,
  };

  const inputProps = {
    id: autoId,
    colorScheme,
    checked,
    ...props,
  };

  const sliderProps = { colorScheme };

  if (!captionUnchecked && !captionChecked) {
    return (
      <Label {...labelProps}>
        <Input {...inputProps} />
        <Slider {...sliderProps} />
      </Label>
    );
  }

  return (
    <Inline space="small" css={{ overflow: 'visible' }}>
      {captionUnchecked && (
        <SwitcherCaption checked={checked}>{captionUnchecked}</SwitcherCaption>
      )}
      <Label {...labelProps}>
        <Input {...inputProps} />
        <Slider {...sliderProps} />
      </Label>
      {captionChecked && (
        <SwitcherCaption checked={!checked}>{captionChecked}</SwitcherCaption>
      )}
    </Inline>
  );
};

export default Switch;
