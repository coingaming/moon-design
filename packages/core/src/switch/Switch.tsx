import React from 'react';
import { uniqueId } from '@heathmont/moon-utils';
import {Text} from "../index";
import Inline from '../inline/Inline';
import {Size} from "./private/Size";
import ButtonLabels from "./styles/ButtonLabels";
import Input from "./styles/Input";
import Label from "./styles/Label";
import Slider from "./styles/Slider";

const switchWidthProperty = '--switch-width';
const switchWidth = `var(${switchWidthProperty})`;

const switchHeightProperty = '--switch-height';
const switchHeight = `var(${switchHeightProperty})`;

type HTMLInputProps = React.InputHTMLAttributes<HTMLInputElement>;

export type SwitchProps = {
  checked?: boolean;
  disabled?: boolean;
  className?: string;
  colorScheme?: boolean;
  id?: HTMLInputProps['id'];
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
  size?: Size;
  isRtl?: boolean;
  buttons?: {
    onLabel: string;
    offLabel: string;
  }
};

const Switch: React.FC<SwitchProps> = ({
  className,
  checked = false,
  disabled = false,
  colorScheme,
  id,
  isRtl,
  size = 'sm',
  buttons,
  ...props
}) => {
  const autoId = id || `Switch-${uniqueId()}`;

  const labelProps = {
    switchHeightProperty,
    switchWidthProperty,
    switchHeight,
    switchWidth,
    className,
    size,
    disabled,
    button: !!buttons
  };

  const inputProps = {
    switchWidth,
    id: autoId,
    colorScheme,
    checked,
    size,
    button: !!buttons,
    isRtl,
    ...props,
    onChange: props.onChange && !disabled ? props.onChange : () => {}
  };

  const sliderProps = {
    colorScheme,
    isRtl,
    checked,
    size,
    button: !!buttons,
    switchHeight,
    switchWidth
  };

  return (
    <Inline
      space="small"
      style={{ overflow: 'visible' }}
      dir={isRtl ? 'rtl' : 'ltr'}
    >
      <Label {...labelProps}>
        <Input {...inputProps} />
        <Slider {...sliderProps}>
          {
            !!buttons && (<ButtonLabels>
              <Text>{buttons.offLabel}</Text>
              <Text>{buttons.onLabel}</Text>
            </ButtonLabels>)
          }
        </Slider>
      </Label>
    </Inline>
  );
};

export default Switch;
