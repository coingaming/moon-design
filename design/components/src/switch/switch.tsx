/** @jsx jsx */
import * as React from 'react';
import { jsx, CSSObject } from '@emotion/core';
import rem from 'polished/lib/helpers/rem';
import { animation, colors } from '@heathmont/sportsbet-tokens';

jsx;

type SwitchProps = React.InputHTMLAttributes<HTMLInputElement> & {};

const SwitchHeight = 32; // px
const SwitchWidth = 64; // px

const switcher: CSSObject = {
  position: 'relative',
  display: 'inline-block',
  flexShrink: 0,
  width: rem(SwitchWidth),
  height: rem(SwitchHeight),
};

const inputStyle: CSSObject = {
  opacity: 0,
  width: 0,
  height: 0,
  '&:checked + .slider': {
    backgroundColor: colors.brand,
  },
  '&:checked + .slider:before': {
    backgroundColor: colors.neutral[10],
    transform: 'translateX(32px)',
  },
};

const slider: CSSObject = {
  position: 'absolute',
  width: rem(SwitchWidth),
  height: rem(SwitchHeight),
  top: 0,
  left: 0,
  right: 0,
  bottom: 0,
  cursor: 'pointer',
  backgroundColor: colors.neutral[70],
  transition: `${animation.speed.default}s`,
  borderRadius: rem(34),
  '&::before': {
    position: 'absolute',
    content: '""',
    height: rem(24),
    width: rem(24),
    left: rem(4),
    bottom: rem(4),
    backgroundColor: colors.neutral[20],
    transition: `${animation.speed.default}s`,
    borderRadius: '50%',
  },
};

const Switch: React.FC<SwitchProps> = props => {
  return (
    /* Disable as `label` wraps input */
    /* eslint-disable-next-line jsx-a11y/label-has-associated-control */
    <label css={switcher}>
      <input css={inputStyle} type="checkbox" {...props} />
      <span css={slider} className="slider" />
    </label>
  );
};

export { Switch };
