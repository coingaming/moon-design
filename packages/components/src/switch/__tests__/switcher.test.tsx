import * as React from 'react';
import { create } from 'react-test-renderer';
import 'jest-styled-components';

import { Switch } from '../switch';

describe('Switch', () => {
  test('renders by default', () => {
    const checkbox = create(<Switch />);

    expect(checkbox).toMatchSnapshot();
  });

  test('renders as checked', () => {
    const checkbox = create(<Switch checked />);

    expect(checkbox).toMatchSnapshot();
  });
});
