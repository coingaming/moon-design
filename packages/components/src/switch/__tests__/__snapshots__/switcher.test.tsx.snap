// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders as checked 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 4rem;
  height: 2rem;
}

.c1 {
  opacity: 0;
  width: 0;
  height: 0;
}

.c1:checked + .slider {
  background-color: #0CD664;
}

.c1:checked + .slider:before {
  background-color: #FFF;
  -webkit-transform: translateX(32px);
  -ms-transform: translateX(32px);
  transform: translateX(32px);
}

.c2 {
  position: absolute;
  width: 4rem;
  height: 2rem;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  cursor: pointer;
  background-color: #31373F;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  border-radius: 2.125rem;
}

.c2::before {
  position: absolute;
  content: "";
  height: 1.5rem;
  width: 1.5rem;
  left: 0.25rem;
  bottom: 0.25rem;
  background-color: #8697A2;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  border-radius: 50%;
}

<label
  className="c0"
>
  <input
    checked={true}
    className="c1"
    type="checkbox"
  />
  <span
    className="c2 slider"
  />
</label>
`;

exports[`Switch renders by default 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 4rem;
  height: 2rem;
}

.c1 {
  opacity: 0;
  width: 0;
  height: 0;
}

.c1:checked + .slider {
  background-color: #0CD664;
}

.c1:checked + .slider:before {
  background-color: #FFF;
  -webkit-transform: translateX(32px);
  -ms-transform: translateX(32px);
  transform: translateX(32px);
}

.c2 {
  position: absolute;
  width: 4rem;
  height: 2rem;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  cursor: pointer;
  background-color: #31373F;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  border-radius: 2.125rem;
}

.c2::before {
  position: absolute;
  content: "";
  height: 1.5rem;
  width: 1.5rem;
  left: 0.25rem;
  bottom: 0.25rem;
  background-color: #8697A2;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  border-radius: 50%;
}

<label
  className="c0"
>
  <input
    className="c1"
    type="checkbox"
  />
  <span
    className="c2 slider"
  />
</label>
`;
